### PR TITLE
#26: Feature/#26 chat

### PR DESCRIPTION
**Pull request by clemenspeters**

### CODE REVIEW

**Comment by nschoen** [comment]
Let's merge the HomeScreen first, then rebase onto the master and then use our custom Alert component, which I updated to support `info` and `success` messages, to show notifications when the app is open.

File: helpers/api.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import fakeApi from './fakeApi';
 
 const BASE_URL = Api.host;
 
-const request = (method, url, bodyParams, queryParams) => {
+const request = (method, url, bodyParams, queryParams, queryHeaders) => {
```
Comment (nschoen) [inline helpers/api.js]: Name queryHeaders, just to header and set a default value via `headers = {}`

File: helpers/api.js [hunk 2]
```diff
@@ -28,6 +28,10 @@ const request = (method, url, bodyParams, queryParams) => {
     options.params = queryParams;
   }
 
+  if (queryHeaders) {
+    options.headers = queryHeaders;
+  }
+
```
Comment (nschoen) [inline helpers/api.js:30]: Remove this and set the header value in the `options` object directly

File: helpers/api.js [hunk 3]
```diff
@@ -36,10 +40,10 @@ const request = (method, url, bodyParams, queryParams) => {
...
 };
 
-const get = (url, queryParams) => request('get', url, null, queryParams);
-const post = (url, bodyParams, queryParams) => request('post', url, bodyParams, queryParams);
-const put = (url, bodyParams, queryParams) => request('put', url, bodyParams, queryParams);
-const patch = (url, bodyParams, queryParams) => request('patch', url, bodyParams, queryParams);
+const get = (url, queryParams) => request('get', url, null, queryParams, {});
+const post = (url, bodyParams, queryParams, queryHeaders) => request('post', url, bodyParams, queryParams, queryHeaders);
+const put = (url, bodyParams, queryParams) => request('put', url, bodyParams, queryParams, {});
+const patch = (url, bodyParams, queryParams) => request('patch', url, bodyParams, queryParams, {});
```
Comment (nschoen) [inline helpers/api.js]: Remove all empty object `{}` params. Rename queryHeaders to just `headers` please.

**Review by nschoen — CHANGES_REQUESTED** [review]

File: App.js [hunk 4]
```diff
@@ -44,6 +47,15 @@ if (UIManager.setLayoutAnimationEnabledExperimental) {
...
 
+async function getiOSNotificationPermission() {
+  const { status } = await Permissions.getAsync(
+    Permissions.NOTIFICATIONS,
+  );
+  if (status !== 'granted') {
+    await Permissions.askAsync(Permissions.NOTIFICATIONS);
+  }
+}
+
```
Comment (calvinschmidt) [inline App.js]: We have methods for this in `helpers/lib.js`. Please use them.

**Review by calvinschmidt — CHANGES_REQUESTED** [review]

File: App.js [hunk 5]
```diff
@@ -44,6 +47,15 @@ if (UIManager.setLayoutAnimationEnabledExperimental) {
 
 chatStore.load();
```
Comment (calvinschmidt) [inline App.js]: This does not work if the user is not logged in and the userStore isn't initialized yet.

File: App.js [hunk 6]
```diff
@@ -78,6 +90,19 @@ class App extends React.Component {
...
+    Notifications.addListener(notification => {
+      if (notification.origin === 'received' && Platform.OS === 'ios') {
+        ReactAlert.alert(notification.title, notification.body);
+      }
+    });
+  };
+
+  componentWillMount() {
+    getiOSNotificationPermission();
+    this.listenForNotifications();
```
Comment (calvinschmidt) [inline App.js]: It already asks for the notification if the user is still logged out. Please check here if the user is logged in.

**Review by calvinschmidt — CHANGES_REQUESTED** [review]

**Review by nschoen — APPROVED** [review]

**Review by calvinschmidt — APPROVED** [review]

**Comment by calvinschmidt** [comment]
Thanks for resolving, looks good now!

**Pull request merged**